### PR TITLE
Put `asserts` on various `assert*` functions in chai.

### PR DESCRIPTION
**Pull request by bmeurer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.chaijs.com/api/assert/

Why?

I'm testing the waters here to see how folks feel about these changes. We're using chai quite extensively in Chrome DevTools and the fact that `assert` functions don't play well with control flow based type narrowing is quite annoying (and we even ended up wrapping the `assert`s into otherwise useless helpers). We believe this should be solved here.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/chai/index.d.ts
1. types/chai/chai-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f2c0d8464fe4823dd254a9bc4d509e7732ef2553
</p>

**Review by weswigham — COMMENTED** [review]
I think this is a good idea, but would like to give the actual type maintainers a bit more time to weigh in before I give this a merge. I'll probably stop by again later in the week, and if things are still silent from the pinged individuals, just merge it.

**Comment by bmeurer** [comment]
> I think this is a good idea, but would like to give the actual type maintainers a bit more time to weigh in before I give this a merge. I'll probably stop by again later in the week, and if things are still silent from the pinged individuals, just merge it.

Thanks for the review. As mentioned, I'm mostly testing the waters here. Happy to expand this to cover more of the `assert` functions if you guys are okay with upstreaming this.

**Review by sheetalkamat — APPROVED** [review]

**Comment by bmeurer** [comment]
Ready to merge

**Pull request merged**
